src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud/config.jelly

```diff
@@ -119,5 +118,9 @@ THE SOFTWARE.
     <f:entry title="${%Defaults Provider Template Name}" field="defaultsProviderTemplate">
       <f:textbox default=""/>
     </f:entry>
+
     <f:optionalProperty field="garbageCollection" title="${%Enable garbage collection}"/>
+
+    <f:descriptorList field="traits" descriptors="${descriptor.allTraits}" instances="${it.traitInstances}" />
```

Comment:
Consider using `<f:repeatableHeteroProperty>` instead and benefit from automatic data-binding.

Reply:
I originally had used `<f:repeatableHeteroProperty field="traits" oneEach="true" addCaption="${%Add trait}" />` but I couldn't figure out how to get the default instances behavior of `<:descriptorList>`. `<f:descriptorList>` ends up calling `DescriptorImpl#getDefaultTraits` the first time through to fill in the defaults. Is there a way to do that with `<f:repeatableHeteroProperty>`?